standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -2419,9 +2419,9 @@ private void create_table_core(final RawStore ms, final CreateTableRequest req)
       }
 
       if (!TableType.VIRTUAL_VIEW.toString().equals(tbl.getTableType())) {
-        if (tbl.getSd().getLocation() == null
-            || tbl.getSd().getLocation().isEmpty()) {
-          tblPath = wh.getDefaultTablePath(db, tbl);
+        if (tbl.getSd().getLocation() == null || tbl.getSd().getLocation().isEmpty()) {
+          String relPath = tbl.getTableName() + (tbl.isSetTxnId() ? "_v" + tbl.getTxnId() : "");
```

Comment:
How is the new location looks like?
```
<WAREHOUSE_ROOT>/<DB_NAME>.db/<TABLE_NAME>_v<TXN_ID>
```

This might get mixed up if there are old tables, or tables created with `hive.txn.nonblocking.droptable.enabled=false` and the name ends with _v1234. Small chance, but...

What about using directories instead, like:
```
<WAREHOUSE_ROOT>/<DB_NAME>.db/<TABLE_NAME>/v_<TXN_ID>
```

Reply:
I think, <TABLE_NAME>/v_<TXN_ID> would be even more confusing + we would need to take care of the <TABLE_NAME> folder if 0 versions remained:
><WAREHOUSE_ROOT>/<DB_NAME>.db/<TABLE_NAME>/v_12/delta_000013_000015
><WAREHOUSE_ROOT>/<DB_NAME>.db/<TABLE_NAME>/delta_00001_000011